### DISCUSSION TITLE
zh/guide/install/reverse-proxy

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#41] [Announcements]
# zh/guide/install/reverse-proxy

AList v3的文档

https://alist.nn.ci/zh/guide/install/reverse-proxy.html

**Comment by RoryBen** [comment]
为啥127.0.0.1:5244无法访问

**Reply by onlybigroc** [reply]
我的也是开放端口无法正常访问

**Comment by ss454534** [comment]
请问如果是Nginx Proxy Manager应当如何设置？

**Reply by ventusoon** [reply]
找到了吗

**Reply by itsdapi** [reply]
一直没弄明白

**Reply by ariesly15** [reply]
同问 Nginx Proxy Manager 设置

**Reply by rational-stars** [reply]
同问

**Reply by ariesly15** [reply]
``` ng
location / {
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_set_header Host $http_host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header Range $http_range;
  proxy_set_header If-Range $http_if_range;
  proxy_redirect off;
  proxy_pass http://IP:PORT;
  client_max_body_size 20000m;
}
```
我用这个目前没问题

**Comment by shiyest** [comment]
IIS反向代理 网盘上传大文件报413，怎么添加代码指定最大上传文件大小？

**Reply by shiyest** [reply]
在网站web.config中configuration字段下添加如下代码：

**Reply by shiyest** [reply]
<system.web>
        <httpRuntime maxRequestLength="214783647" />
    </system.web>
    <system.webServer>
        <security>
            <requestFiltering>
                <requestLimits maxAllowedContentLength="214783647" />
            </requestFiltering>
        </security>
    </system.webServer>

**Reply by shiyest** [reply]
iis允许上传文件长度最大为2G，上述代码中已添加为最大，再大就会报500错误！

**Comment by Rosewives** [comment]
[CyberPanel](https://docs.litespeedtech.com/lsws/cp/cpanel/rewrite-proxy/) 里应当如何设置 Alist

**Comment by czy20010112** [comment]
您好 请问我在centos用一键脚本安装后 采用nginx反代 设置后访问公网ip的我网站显示System error: TypeError: Failed to fetch dynamically imported module: https://域名/assets/Layout.c75b132a.js ，请问这种情况这么解决啊？ 按照您文档中的删除了nginx默认的lozation及其{}内的内容 （已经在config文件中填写了我的site_url,采用的https://域名的形式（没指定端口））

**Reply by czy20010112** [reply]
site_url写成http后能进去了但是只在标签页那显示图标和名称...网站内容一片空白

**Reply by czy20010112** [reply]
改成https刷新了几次进去了...很怪 但还是偶尔出现这个报错

**Reply by zxcv-12345** [reply]
他这个有时候就是比较玄学

**Comment by Yofo-me** [comment]
如何子目录反代？

**Reply by PunchlY** [reply]
```
# data/config.json
{
    ...
    "site_url": "/alist",
    ...
}
```
```
# nginx
location ^~ /alist/ {
    proxy_pass http://127.0.0.1:5244/;
}
```

**Reply by anglebai** [reply]
设置后，反代访问正常；原始地址访问一片空白

**Reply by Mai0313** [reply]
同問 反向代理訪問正常，但一片空白

**Comment by 3135093380** [comment]
密码咋看的？

**Reply by zxcv-12345** [reply]
Alist 安装成功！

访问地址：http://YOUR_IP:5244/

配置文件路径：/opt/alist/data/config.json
$查看管理员信息，请执行
cd /opt/alist
./alist admin

查看状态：systemctl status alist
启动服务：systemctl start alist
重启服务：systemctl restart alist
停止服务：systemctl stop alist

温馨提示：如果端口无法正常访问，请检查 服务器安全组、本机防火墙、Alist状态

**Reply by zxcv-12345** [reply]
查看管理员信息那边可以显示密码

**Comment by lostornot** [comment]
Cannot read properties of undefined (reading 'status')
nginx 限制容量也改了，也试了caddy，都是在上传时有这个问题，还会碰到413. 但是caddy本身不限制容量，nginx也改成大容量了，还是有这个问题。

**Reply by RenMu-V** [reply]
对，我也遇到了，好烦哦caddy本身就不要配置啥。不知道alist底层配置文件是不是有要需要修改的地方？
[![imagec8af891d9a0db32b.png](https://img.yaomomo.com/images/imagec8af891d9a0db32b.png)](https://img.yaomomo.com/image/8S3n)
不是413就是status

**Comment by zxcv-12345** [comment]
他这个alist我觉得不太适合传大文件的东西上去

Deeply Mu ***@***.***> 于 2023年3月18日周六 05:35写道：

**Comment by roystool** [comment]
开启反代之后，视频加载错误怎么解决呢？

**Comment by xiaoaqiu** [comment]
桌面版突然用不了啦

**Comment by lizhiwei8061** [comment]
求助一个 nginx 反代理https的配置，我使用的域名访问，已经申请证书

**Comment by GemkingYum** [comment]
我根据以上配置信息配置nginx的反向代理后，上传文件时稳定报错Request failed with status code 500
切换端口登陆，不经过反向代理，就可以上传成功，请问有人遇到同样的问题吗

**Reply by xXiaoXiang** [reply]
应该是你nginx所在的磁盘没有空间了，在 nginx.conf 里添加一行 client_body_temp_path /tmp/nginx 1 2; 试试

**Comment by zlf017** [comment]
如何给该项目设置个项目命名，如: 通过`/alist `访问主页

**Comment by 17gather** [comment]
nginx: [emerg] /etc/nginx/nginx.conf:32 中不允许使用“proxy_pass”指令

**Comment by yqfcpy** [comment]
443端口转5244 大白屏

**Comment by ashuai** [comment]
能在nginx反向代理的时候自动帮baidu把ua加上吗？

**Comment by se7enmuting** [comment]
反代用的群晖6内置的反代，然后参考这里的说明 site_url 设置了 https://****，然后所有平台的浏览器就无法播放视频了，删掉  site_url 的设置，立马浏览器可以播放视频。3.30.0版本。

**Comment by RedSTARO** [comment]
反代后访问白屏

**Comment by chenyang137** [comment]
反向代理添加https证书后访问一切正常，但是https协议的webdav无法使用。浏览器登陆提示Method Not Allowed   客户端提示连接失败  版本3.34.0

**Comment by SunRain** [comment]
参照https://github.com/alist-org/alist/issues/726,使用NginxProxyManager作为反代服务的似乎,似乎需要做一些修改(以下复制黏贴自github的issue内容)

目前最新的alist版本,在alist的config.json里面设置site_url的时候,如果不加上上述使用的非标端口,即使按照这个设置配置NginxProxyManager的反代,似乎也不起作用.必须要在alist的config.json里面设置site_url+端口的形式才行(类似site_url": "https://domain.com:port"这样)

**Reply by SunRain** [reply]
https://github.com/alist-org/alist/issues/726
